### ISSUE TITLE
Writing a non trainable layer Reshape Layer in Keras

### ISSUE THREAD

**Issue by MehulAjax21** [fchollet/keras#6527]
Hi guys, 

I have developed a numpy function for custom reshaping required to convert a block of 3x3 pixel into a timestep to feed into an LSTM.

This function is working correctly on numpy arrays.

Now I want to make my network end to end trainable. For that, I will have to write this layer as keras layer. Can I use the numpy function which I wrote directly?

Or will I have to write it using theano ( My backend)?

**Comment by kgrm** [comment]
You could use keras.backend functions to preserve compatibility with theano and tensorflow. But you can't use numpy code in keras layers, since it relies on the backend for differentiation.

But I should also note, there exists a `Reshape` layer. What use case does it fail to cover?

**Comment by MehulAjax21** [comment]
"But you can't use numpy code in keras layers, since it relies on the backend for differentiation"

But this layer will be non trainable. I dont need to differentiate ut. So I was thinking if I can use numpy code or not.


"But I should also note, there exists a Reshape layer. What use case does it fail to cover?"

I need to divide my data into timesteps to feed the output of my CNN to an LSTM.

My output will be of shape ( none, 1, 224, 224) (theano dim). Now from thr 224 x 224 features, i need to create timesteps by sliding a 3x3 window over the features. These will be reshaped internally so that, it can be fed into LSTM.
The shape will be (none, 5625, 9) ( By zero padding the image to fit sliding window).


My numpy code is doing this exactly. But I can not make the training end to end as The code is not a layer.

**Comment by kgrm** [comment]
Regardless of whether the layer is trainable, it needs to take a backend tensor as an input and produce one as output, and you can't do that with numpy code.

**Comment by MehulAjax21** [comment]
Hi kgrm,
Thanks for this clarification. I can close this issue now. I will try to convert my code using backend methods.

Regards,
Mehul Saxena
